euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/Join.java

```diff
@@ -436,19 +514,21 @@ public boolean isOuter() {
     Union<Either<LEFT, RIGHT>> union =
         new Union<>(name, flow, leftMap.output(), rightMap.output());
 
-    ReduceStateByKey<Either<LEFT, RIGHT>, KEY, Either<LEFT, RIGHT>, OUT, JoinState, W>
+    ReduceStateByKey<Either<LEFT, RIGHT>, KEY, Either<LEFT, RIGHT>, OUT, StableJoinState, W>
         reduce = new ReduceStateByKey(
-              getName() + "::ReduceStateByKey",
-              flow,
-              union.output(),
-              keyExtractor,
-              e -> e,
-              getWindowing(),
-              getEventTimeAssigner(),
-              JoinState::new,
-              new StateSupport.MergeFromStateMerger<>(),
-              partitioning
-        );
+        getName() + "::ReduceStateByKey",
+        flow,
+        union.output(),
+        keyExtractor,
+        e -> e,
+        getWindowing(),
+        getEventTimeAssigner(),
+        (StorageProvider storages, Context ctx) ->
```

Comment:
Could we make this `Context` here `Optional` instead of passing `null`? This is part of client API, ale I think it would be a little cleaner.